dev/breeze/src/airflow_breeze/utils/projects_google_spreadsheet.py

```diff
@@ -0,0 +1,253 @@
+            creds = flow.run_local_server(port=0)
+            # Save the credentials for the next run
+        token_path.write_text(creds.to_json())
+    service = build("sheets", "v4", credentials=creds)
+    sheets = service.spreadsheets()
+    return sheets
+
+
+def get_sheets(json_credentials_file: Path) -> Resource:
+    token_path = Path.home() / ".config" / "gsheet" / "token.json"
+    sheets = authorize_google_spreadsheets(json_credentials_file, token_path)
+    return sheets
+
+
+def write_sbom_information_to_google_spreadsheet(
+    sheets: Resource,
+    docs: dict[str, str],
+    google_spreadsheet_id: str,
+    all_dependencies: list[dict[str, Any]],
+    fieldnames: list[str],
+    include_opsf_scorecard: bool = False,
+):
+    # Use only interesting values from the scorecard
+    cell_field_names = [
+        fieldname
+        for fieldname in fieldnames
+        if fieldname in INTERESTING_OPSF_SCORES or not fieldname.startswith("OPSF-")
+    ]
+
+    num_rows = update_field_values(all_dependencies, cell_field_names, google_spreadsheet_id, sheets)
+    if include_opsf_scorecard:
+        get_console().print("[info]Updating OPSF detailed comments.")
+        update_opsf_detailed_comments(
+            all_dependencies, fieldnames, num_rows, google_spreadsheet_id, docs, sheets
+        )
+
+
+def update_opsf_detailed_comments(
+    all_dependencies: list[dict[str, Any]],
+    fieldnames: list[str],
+    num_rows: int,
+    google_spreadsheet_id: str,
+    docs: dict[str, str],
+    sheets: Resource,
+):
+    opsf_details_field_names = [
+        fieldname for fieldname in fieldnames if fieldname in INTERESTING_OPSF_DETAILS
+    ]
+    start_opsf_column = fieldnames.index(opsf_details_field_names[0]) - 1
+    opsf_details = []
+    opsf_details.append(
+        {
+            "values": [
+                {"note": docs[check]}
+                for check in INTERESTING_OPSF_FIELDS
+                if check != INTERESTING_OPSF_FIELDS[0]
+            ]
+        }
+    )
+    get_console().print("[info]Adding notes to all cells.")
+    for dependency in all_dependencies:
+        note_row = convert_sbom_dict_to_spreadsheet_data(opsf_details_field_names, dependency)
+        opsf_details.append({"values": [{"note": note} for note in note_row]})
+    notes = {
+        "updateCells": {
+            "range": {
+                "startRowIndex": 1,
+                "endRowIndex": num_rows + 1,
+                "startColumnIndex": start_opsf_column,
+                "endColumnIndex": start_opsf_column + len(opsf_details_field_names) + 1,
+            },
+            "rows": opsf_details,
+            "fields": "note",
+        },
+    }
+    update_note_body = {"requests": [notes]}
+    get_console().print("[info]Updating notes in google spreadsheet.")
+    sheets.batchUpdate(spreadsheetId=google_spreadsheet_id, body=update_note_body).execute()
+
+
+def calculate_range(num_columns: int, row: int) -> str:
+    import string
```

Comment:
Not sure we need a local import for that

Reply:
Yes. could be regular import..